### ISSUE TITLE
Failed to add mined block to chain with reason Invalid spend

### ISSUE THREAD

**Issue by gjfrn** [iron-fish/ironfish#619]
I have successfully mined and added 3 blocks so far, but my fourth one has failed due to 'Invalid spend':

Successfully mined block 00000000036e8cc05a70361127a9e52ab25708b6b1953550f1ec03a16b75f036 (12313) has 11 transactions
Failed to add mined block to chain with reason Invalid spend

**Comment by gjfrn** [comment]
Appears to be a different issue than https://github.com/iron-fish/ironfish/pull/393.

**Comment by gjfrn** [comment]
I have had this happen for the next two blocks that I mined:
Successfully mined block 0000000006103af6a808c892723c512d885ca85cf8789f76f74600e0c3353807 (12340) has 4 transactions
Failed to add mined block to chain with reason Invalid spend

Successfully mined block 000000000b46f32e950678b7c2948e8de66a5ade4ef2a0b62e3bfcedd7ba57e4 (12494) has 5 transactions
Failed to add mined block to chain with reason Invalid spend

**Comment by gjfrn** [comment]
Two more examples today:
Successfully mined block 0000000006f8e7df22aa2d90f69a5b38474cbb34b55480a87602a8e6db4f700d (13550) has 3 transactions
Failed to add mined block to chain with reason Invalid spend
Successfully mined block 0000000004dbdff138f3ba6dff62b21eb690f3ec92c559b73af785d39172bed8 (13846) has 2 transactions
Failed to add mined block to chain with reason Invalid spend

**Comment by gjfrn** [comment]
Seems there are a number of other users reporting this recently since my original report so definitely an ongoing issue.

**Comment by NullSoldier** [comment]
Users are reporting a suspicious amount of invalid add blocks with invalid spends. See ther example below. In one case, a stuck block that took 4 hours to mine had this issue which was very frustrating because it took a long time to mine.


```
Successfully mined block 0000000005afd3ac33be1082740dd909d7261d5816c76438490befb43489fc33 (13561) has 2 transactions
Failed to add mined block to chain with reason Invalid spend
Successfully mined block 0000000008b696906d12909a6049e7779e544eff1ea5c56db03c30e33f1c9599 (13561) has 2 transactions
Failed to add mined block to chain with reason Invalid spend
```

**Comment by 1frkmtk1** [comment]
Same problem
The problem appeared after updating to version 0.1.9.
Graffiti: frkmtk

Dec 10 08:05:20 Ubuntu-2004-focal-64-minimal yarn[791133]: Successfully mined block 0000000001062614075264a21fb02dda811c0a05e95fbd74a4eb144377944b67 (15967) has 6 transactions
Dec 10 08:05:22 Ubuntu-2004-focal-64-minimal yarn[791133]: Failed to add mined block to chain with reason Invalid spend

**Comment by gjfrn** [comment]
4/4 failures today:
Successfully mined block 0000000001861d526e0a8bc03fa87c1c244faa47f8450b3fac600856bc3a4583 (15887) has 2 transactions
Failed to add mined block to chain with reason Invalid spend

Successfully mined block 000000000d5cce484ae6720bf7e8f247205d13f5787603c1f5d9c5e62e2d8145 (15967) has 6 transactions
Failed to add mined block to chain with reason Invalid spend
Successfully mined block 000000000063132e4999bbb8832c8ba68b383caaa64f6cd007b2066935ef99f7 (15967) has 6 transactions
Failed to add mined block to chain with reason Invalid spend

Successfully mined block 000000000697f1a8f5d3c10e145b393858532c46c7bb72a3ae883d66ff3f3bb5 (16025) has 3 transactions
Failed to add mined block to chain with reason Invalid spend

Brings my failure rate to 50%!

Note: this started occurring for me before 0.1.9, though possibly it's been worse ever since.

**Comment by 1frkmtk1** [comment]
One more block is not counted.
And I minted the block 2 hours earlier than the other miner, according to the explorer. Fork?

My journal:
Dec 10 14:16:05 Ubuntu-2004-focal-64-minimal yarn[791133]: Successfully mined block 000000000d79ee8be713f761a569052fe45f1e7a42ff8409067c89ad068fdd3c (16156) has 6 transactions
Dec 10 14:16:07 Ubuntu-2004-focal-64-minimal yarn[791133]: Failed to add mined block to chain with reason Invalid spend

Explorer:
Timestamp
10/12/2021 16:16:44

0/4 blocks after 0.1.9 upd

**Comment by ngoldbla** [comment]
I am encountering double spend error once / and invalid spends three times

Version       0.1.9 @ cca3c88
Node Name     NONE
Peer Identity xhkXp6fp3JEjdzbqrJkY/n5A7gshFPWz5TO0lQFJwFs=
Peer Agent    if/cli/cca3c88
Peer Port     9033
Bootstrap     test.bn1.ironfish.network
 
WebSocket server started at :::9033
Connected to the Iron Fish network
Starting sync from lFz0Fdj (whittygooglecloud60). work: +36100674237946, ours: 15591, theirs: 15769
Finding ancestor using linear search on last 3 blocks starting at 00000...f5215 (15591) from peer lFz0Fdj (whittygooglecloud60) at 15769
Found peer lFz0Fdj (whittygooglecloud60) ancestor 00000...f5215, syncing from 15591 -> 15769 (178) after 1 requests
...
Finished syncing 178 blocks from lFz0Fdj (whittygooglecloud60)
...
Added block to fork seq: 15853, head-seq: 15853, hash: 00000...9bcfc, head-hash: 00000...135b4, work: 1219309880735037, head-work: 1219309975820227, work-diff: 95085190
Successfully mined block 00000000041fe3ca016518a1d6ee092ef83adc365ebf505b600c9c082eb3e039 (15856) has 2 transactions
Failed to add mined block to chain with reason Invalid spend
Added block to fork seq: 15857, head-seq: 15857, hash: 00000...3c28a, head-hash: 00000...9c828, work: 1219681623475128, head-work: 1219681623475128, work-diff: 0
...
Added block to fork seq: 15953, head-seq: 15953, hash: 00000...2ef5a, head-hash: 00000...b6ffb, work: 1227183166244368, head-work: 1227183166244368, work-diff: 0
Not connected to the Iron Fish network
Connected to the Iron Fish network
Peer ekHAhV7 sent orphan 00000...f6543 (15966), syncing orphan chain.
Starting sync from ekHAhV7. work: +399040925462, ours: 15959, theirs: 15965
...
Finished syncing 53 blocks from Lk3nZQd (bombafish), skipped 46
...
Added block to fork seq: 15943, head-seq: 16014, hash: 00000...5e408, head-hash: 00000...db17f, work: 1226518118487358, head-work: 1230683250941738, work-diff: 4165132454380
Successfully mined block 00000000030de4727c59a0208613f39fbd908a5d2195ef10dda89c0367a35546 (16018) has 3 transactions
Failed to add mined block to chain with reason Invalid spend
...
Added block to fork seq: 16041, head-seq: 16045, hash: 00000...587b9, head-hash: 00000...74e73, work: 1231960209938905, head-work: 1232144756977392, work-diff: 184547038487
Successfully mined block 0000000004e0c670f8dcd1f36e69b547a6d845fb6373aeda0151caf79d433a4e (16048) has 8 transactions
Failed to add mined block to chain with reason Double spend
Added block to fork seq: 16049, head-seq: 16049, hash: 00000...fbdd9, head-hash: 00000...f4775, work: 1232326471437129, head-work: 1232326493570256, work-diff: 22133127
...
Reorganizing chain from 00000...3e637 (16079) for 00000...f8deb (16079) on prev 00000...70b2a (16078)
Reorganized chain. blocks: 1, old: 00000...3e637 (16079), new: 00000...70b2a (16078), fork: 00000...70b2a (16078)
Added block seq: 16080, hash: 00000...7bb7c, txs: 2, progress: 100.00%, time: 3722.9ms
...
Added block seq: 16200, hash: 00000...dd6a2, txs: 1, progress: 100.00%, time: 120.6ms
Successfully mined block 0000000004d041f07911f997693e936c9651bd3fe308a1583fd2b54d4ce70305 (16212) has 2 transactions
Failed to add mined block to chain with reason Invalid spend
Added block to fork seq: 16217, head-seq: 16217, hash: 00000...cf26b, head-hash: 00000...2dbe0, work: 1239841488286730, head-work: 1239841531933296, work-diff: 43646566

**Comment by Kokorin9** [comment]
Same issue here
Successfully mined block 00000000145d93b4eadb5a27d752dea1f565df69f7e83fd4a6a737e178a3fb49 (16343) has 2 transactions

Failed to add mined block to chain with reason Invalid spend

**Comment by jTiKey** [comment]
Same issue here.
Discord "support," said it's intended this way.

**Comment by marutyan** [comment]
Dec 09 22:31:40 ubuntu yarn[1279754]: Successfully mined block 00000000034bf9b550ebc223706bd4d151abd186fc655a4174d641236c6d3b71 (15860) has 2 transactions
root@ubuntu:~# ironfish chain:block 00000000034bf9b550ebc223706bd4d151abd186fc655a4174d641236c6d3b71
yarn run v1.22.17
$ yarn build && yarn start:js chain:block 00000000034bf9b550ebc223706bd4d151abd186fc655a4174d641236c6d3b71
$ tsc -b
$ cross-env OCLIF_TS_NODE=0 IRONFISH_DEBUG=1 node bin/run chain:block 00000000034bf9b550ebc223706bd4d151abd186fc655a4174d641236c6d3b71
No block found with hash 00000000034bf9b550ebc223706bd4d151abd186fc655a4174d641236c6d3b71

**Comment by i1off-tape** [comment]
: Successfully mined block 0000000000ede84f089e9c3651515e66ec962c6e4fca05ea319019f652f8795b (18759) has 7 transactions: :Failed to add mined block to chain with reason Invalid spend

too

**Comment by dguenther** [comment]
This should have been addressed as of 0.1.12. Please file a new issue if you're still seeing this issue, as the cause will be a different one. Thanks for reporting!
